src/test/java/com/segment/analytics/android/integrations/amplitude/AmplitudeTest.java

```diff
@@ -363,7 +363,7 @@ public void trackWithTotalV2() {
 
     ArgumentCaptor<Revenue> firstArgument = ArgumentCaptor.forClass(Revenue.class);
     verify(amplitude).logRevenueV2(firstArgument.capture());
-    assertThat(expectedRevenue.equals(firstArgument));
+    assertThat(expectedRevenue).isEqualToComparingFieldByFieldRecursively(firstArgument.getValue());
```

Comment:
Do we need this variant for the revenue field as well?

Reply:
You're right, I could check with something like: 

```
expectedRevenue = new Revenue().setProductId("bar")
        .setPrice(2)
        .setQuantity(10)
        .setReceipt("baz", "qux")
        .setEventProperties(properties.toJsonObject());

verify(amplitude).logRevenueV2(expectedRevenue);
```

 I'll update this PR to simplify the tests.